File: android/app/src/main/kotlin/com/devlabs/woosim_printer_flutter/printer/PrinterDevice.kt
package com.devlabs.woosim_printer_flutter.printer

data class PrinterDevice(
    val name: String,
    val address: String
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "name" to name,
            "address" to address
        )
    }
}

File: lib/models/printer_device.dart
class PrinterDevice {
  const PrinterDevice({
    required this.name,
    required this.address,
  });

  final String name;
  final String address;

  factory PrinterDevice.fromMap(Map<dynamic, dynamic> map) {
    return PrinterDevice(
      name: map['name']?.toString() ?? 'Unknown Printer',
      address: map['address']?.toString() ?? '',
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'name': name,
      'address': address,
    };
  }
}
